### DISCUSSION TITLE
[Button] Add Line Height feature.

### DISCUSSION THREAD

**Discussion by klootas** [godotengine/godot-proposals#10476] [GUI]
Now that Button supports autowrap (https://github.com/godotengine/godot/pull/87025 👍), it would be useful to be able to specify the line height of the wrapped label. As I was looking for it, I expected to find it under Theme Overrides/Constants.

**Comment by timothyqiu** [comment]
See also https://github.com/godotengine/godot-proposals/issues/9778

**Reply by timothyqiu** [reply]
@bruvzg Asking here as the question is not very related to the line spacing PR.

According to your explaination of how `line_spacing` works, it seems that currently the decender line is touching the ascender line of the text below by default (line spacing is zero).

As far as I know, fonts have their preferred line spacing information built in. e.g., Freetype has `height` in size metrics representing the natrual distance between baselines. Should we use this in some way when rendering multiline text? For example, make line spacing a percentage of `height` so that it's compatible with different font sizes.

**Reply by bruvzg** [reply]
> it seems that currently the decender line is touching the ascender line of the text below by default 

Yes.

> As far as I know, fonts have their preferred line spacing information built in. e.g., Freetype has height in size metrics representing the natrual distance between baselines.

There's `height` in the font metrics, and it's usually either equal or smaller than `ascent + desecnt`, so we can use it to calculate default spacing, but not sure how useful it would be. Currently, we use processed line `ascent + descent` (calculated from all used fonts, and adjusted to fit offset glyphs, like diacritics) and not values from the font.
